public/js/pimcore/settings/user/user/settings.js

```diff
@@ -301,6 +301,43 @@ pimcore.settings.user.user.settings = Class.create({
             }
         });
 
+        const validLocales = [
+            ['', '(system)']
+        ];
+        const allLocales = this.data.validLocales ?? [];
+        // Rely on supportedLocalesOf to exclude any non-supported locales
+        Intl.DateTimeFormat.supportedLocalesOf(Object.keys(allLocales), {localeMatcher: "lookup"}).forEach(function (locale) {
+           validLocales.push([locale, allLocales[locale]])
+        });
+
+        const localesStore = new Ext.data.ArrayStore({
+            fields: ['code', 'text'],
+            data : validLocales
+        });
+
+        generalItems.push({
+            xtype: 'combo',
+            fieldLabel: t('datetime_locale'),
+            typeAhead: true,
+            value: this.currentUser.datetimeLocale ?? '',
+            mode: 'local',
```

Comment:
Let's add a `queryMode: 'local'` here as this improves the behaviour of the typeahead search. I'm not sure, but does the `mode` option even exist or should it be replaced with `queryMode`?

Reply:
indeed it's not working, fixed on https://github.com/pimcore/admin-ui-classic-bundle/pull/554/commits/2534755886aaedf09c44d8378197760cbc28e1fa 
fixed also from where i have copied from, but should revisit all the occurrence/usage of this outdated `mode` in a separate PR

Reply:
Wil have a deeper look, opened an issue for it https://github.com/pimcore/admin-ui-classic-bundle/issues/569